frontend/workflows/projectSelector/src/project-selector.tsx

```diff
@@ -57,6 +57,19 @@ const StyledProgressContainer = styled.div({
   },
 });
 
+// Determines if every project has projectData (i.e. the effect has finished fetching the data)
+const allPresent = (state: State): boolean => {
+  let ret = true;
+  // We could potentially check all groups but it's really not necessary since an upstream cannot be added manually.
+  _.forEach(Object.keys(state[Group.PROJECTS]), p => {
+    if (!(p in state.projectData)) {
+      ret = false;
+    }
+    return ret; // Will stop iteration early if false encountered.
```

Comment:
Wont this only iterate once and then return? Also can we remove the usage of `ret` and just return when necessary?

Reply:
nope when it returns true iteration continues. can't return from the closure, need an outer scoped var to track result.

Reply:
Ah ok i see now thanks.